src/Common/MetricHelper.cs

```diff
@@ -50,13 +50,13 @@ public class MetricHelper
         [ThreadStatic]
         private static string _internalCalledCmdlets = "";
 
-        public static string TelemetryId { get=> _telemetryId; set { _telemetryId = value; } }
+        public static string TelemetryId { get => _telemetryId ?? ""; set { _telemetryId = value; } }
```

Comment:
To my understanding, if _internalCalledCmdlets is not initialized, its default value would be null. But we prefer empty string than null, that's the reason you made this change, right?

Reply:
> To my understanding, if _internalCalledCmdlets is not initialized, its default value would be null. But we prefer empty string than null, that's the reason you made this change, right?

You are right. The issue here is I used to think assignment in line#51 will happen in all the threads, but the assignment only happen in the first thread.